docs/mobile-apps/automated-testing/appium/real-devices.md

```diff
@@ -2,190 +2,209 @@
+*  [`carrierConnectivityOnly`](/dev/test-configuration-options/#carrierconnectivityonly)
+*  [`cacheId`](/dev/test-configuration-options/#cacheid)
+*  [`noReset`](/dev/test-configuration-options/#noreset)
+*  [`recordDeviceVitals`](/dev/test-configuration-options/#recorddevicevitals)
+*  [`crosswalkApplication`](/dev/test-configuration-options/#crosswalkapplication)
+*  [`autoGrantPermissions`](/dev/test-configuration-options/#autograntpermissions)
+*  [`enableAnimations`](/dev/test-configuration-options/#enableanimations)
 
-#### **Checking the Appium Version for Your Test**
-1. Log in to Sauce Labs.
-2. Go to **Test Details**.
-3. Find and select the test that you ran using Appium.
-4. Click the **Metadata** tab.
-5. Look for the **Logs** row and select **Appium Log**. The first line should indicate the Appium version. For example, `2019-05-05T17:45:07.541Z - info: Welcome to Appium v1.10.1`.
 
-#### Setting the `browserName`
+### Using `cacheId` and `noReset`
 
-When testing a native mobile app, the value for `browserName` is an empty string, as in `caps.setCapability("browserName", "");`.
+By default, every time you complete a test session, the real device cloud uninstalls your app, performs device cleaning, and de-allocates the device. If you're running multiple tests on the same device, this is inconvenient and inefficient:
 
-#### Setting the Location of Your Mobile App
+* You must to wait for the cleaning process to complete between every test.
+* You lose time in your testing while your app gets reinstalled to the same device each time.
+* There is a small chance that the device could get allocated to another tester before your next test picks it up.
 
-If the app you want to test has been uploaded to a location other than our App Storage, you need to specify this location for `app`, and make sure that this location is accessible to Sauce Labs browsers. For example, `caps.setCapability("app","storage:filename=mapp.zip");`.
+To optimize device availability, consistency, and efficiency for multiple tests, assign a `cacheId` to your tests, which keeps the device allocated to you for 10 seconds after each test completes and skips the allocation and device cleaning process if you immediately start another test. The application and its data will still be uninstalled and reinstalled for the next test, however.
 
-#### Setting the `automationName` for Android Apps
+```js
+caps.setCapability("cacheId", "jnc0x1256");
+```
 
-If you're testing a native mobile app against Android versions 4.0-4.1, or a hybrid mobile against Android versions 4.0 - 4.2, you need to set the capability `"automationName","selendroid"`.
+To skip the uninstallation and reinstallation of your app from the device, you can set `noReset` to `true` in conjunction with using a `cacheId`. This setting adds efficiency, but may not be suitable for test setups that require the application's state to be reset between tests.
```

Comment:
I'm not sure if this is 100% correct, AFAIK `cacheId` keeps the session alive for 10 seconds, if a session is requested with exactly the same caps it will resume in the last state. `noReset` doesn't help with cleaning the app data and or remove it.

But as mentioned, I need to check this, maybe until that time remove this from the docs?

Reply:
I've tested this and you're correct:

- when `cacheId` is set and `noReset` is `false` the app will be removed and reinstalled
- when `cacheId` is set and `noReset` is `true` the app will not be removed and will keep it's previous state